### PR TITLE
Feature/overview

### PR DESCRIPTION
**Pull request by magjo21**

### CODE REVIEW

File: pages/[review].js
Comment (Long-HaHoang) [inline pages/[review].js:24]: Simple and effective. 👍

File: pages/[review].js
Comment (Long-HaHoang) [inline pages/[review].js:64]: You might refactor it later on. Just in case you want to add more aspects or change aspects.
But for now, it's simple and does what it should do.

File: pages/[review].js
Comment (Long-HaHoang) [inline pages/[review].js:75]: You usually place the `_` to the front for private variables.
`_aspect` instead of `aspect_`
I might implement this kind of map ternary operator in my code, later.
Works great if the condition isn't too complex.

File: pages/[review].js
Comment (Long-HaHoang) [inline pages/[review].js:95]: Instead of an unsemantic `div` you can use a `fieldset` to group your inputs.
Depends on what you what to do with div later on with styling.

File: pages/index.js
Comment (Long-HaHoang) [inline pages/index.js:16]: Isn't there a setter function in jotai?
You can look into the Zustand workshop it's a similar concept about setters and getters
It's easier to set/get those aspects later on. 
It's just the complexity increases with more with so many aspects.
You have to add 18 setters to modify each aspect and 18 getters to retrieve them

File: pages/overview.js
Comment (Long-HaHoang) [inline pages/overview.js:10]: typo `litke` instead of `like`

**Review by Long-HaHoang — APPROVED** [review]
Look good to me.
Overall a simple feature and it's easy to read.
Just one thing.
There is a `console.log` somewhere I couldn't find it.

File: pages/[review].js
Comment (Long-HaHoang) [inline pages/[review].js:39]: Please remove this console.log

File: pages/[review].js
Comment (Long-HaHoang) [inline pages/[review].js:71]: Here too.

**Review by Long-HaHoang — COMMENTED** [review]
Ok, I found the console.log

File: pages/[review].js
Comment (F-Kirchhoff) [reply pages/[review].js:24]: instead of this triple AND operation you could use the .includes method:
```js
if( !['good','neutral','bad'].includes(review) ){ ... }
```

File: pages/[review].js
Comment (F-Kirchhoff) [inline pages/[review].js:63]: you are always using previous.good. here. This will lead to bugs if I get your data structure right. I should use `previous[review].aspects`

Comment (F-Kirchhoff) [inline pages/[review].js:63]: here you should should create this dynamically with a map as well as you did in your form!
```js
const newAspects = Object.fromEntries(
  initialAspects.map(aspcect => {
     const oldValue = previous[review].aspects[aspect.name]
     return [ aspect.name, data[aspect.name] ? oldValue + 1 : oldValue ]
  }
)
```
Then use this newAspects in your updated state. This might be a bit more complex but I think you can handle it. There is only one catch. You need the initialAspects names match exactly the keys of your aspects. Please change this accordingly.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Magnus, this looks quite clever I like it a lot! I added some change requests, have a look!

**Comment by magjo21** [comment]
> Hey Magnus, this looks quite clever I like it a lot! I added some change requests, have a look!

Hello Felix,
thanks for your opinion, I will have a look at it.
but first I would like to get a little further in the code.
would be great if you could give me the aprove

**Comment by F-Kirchhoff** [comment]
Hey Magnus, 
I think these changes are necessary right now. I already gave you 90% of the solution. I think you will be done with it fast.

**Comment by magjo21** [comment]
> Hey Magnus, I think these changes are necessary right now. I already gave you 90% of the solution. I think you will be done with it fast.

Hello,
I get the point, I have also already taken 2 of your comments so I would like to leave it at that.

**Comment by F-Kirchhoff** [comment]
Hey Magnus,
I get it that you want to continue on your other interesting features. But change requests from Coaches are not optional. I won't approve this PR if only part of the changes are implemented.
Thank you for understanding.

**Comment by magjo21** [comment]
> Hey Magnus, I get it that you want to continue on your other interesting features. But change requests from Coaches are not optional. I won't approve this PR if only part of the changes are implemented. Thank you for understanding.

Hey Felix,
thank you for your suggestion.
I have implemented it as well as possible.
If you have any other changes let me know.✌️

**Review by NikRoe — APPROVED** [review]
Hi there,

looking good 💪 

Weiter so!

**Review by F-Kirchhoff — COMMENTED** [review]
Thats a very good solution. LGTM 🦊

**Pull request merged**
